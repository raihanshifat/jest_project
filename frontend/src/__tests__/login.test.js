import { render,cleanup, fireEvent ,act} from '@testing-library/react';
import { waitFor } from '@testing-library/dom';
import  axiosInstance from "../axiosinstance"
import axios from 'axios';

// jest.mock("axios")

// axios.create.post =jest.fn(() => Promise.resolve())
// axios.create.interceptors.response.use=jest.fn()
 
// jest.mock('axios', () => {
//     return {
//       create: jest.fn(() => ({
//         post: jest.fn(() => Promise.resolve()),
        // interceptors: {
        //   request: { use: jest.fn(), eject: jest.fn() },
        //   response: { use: jest.fn(), eject: jest.fn() }
        // }
//       }))
//     }
//   })


import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import SignIn from "../components/login";
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme from 'enzyme'

Enzyme.configure({ adapter: new Adapter() });

beforeEach(()=>cleanup())

it("Login page rendered succesfully",()=>{
    const tree=renderer.create(<BrowserRouter><SignIn></SignIn></BrowserRouter>)
    expect(tree).toMatchSnapshot()
})

it("Handle submit working correctly",async ()=>{
    jest.spyOn(axiosInstance, 'post')
    axiosInstance.post = jest.fn(()=>Promise.resolve())
    // const mockObj=jest.spyOn(axios,'post').mockResolvedValue(Promise.resolve())
    // jest.mock('axios', () => {
    //     return {
    //       post: jest.fn()
    //     };
    //   });
    // axiosInstance=jest.fn().mockImplementation(()=>20)
    // const resp = { data: [{ name: 'FooBar' }]};
    // axios.post.mockImplementation(() => Promise.resolve(resp)); 
    
    // axios.create.post = jest.fn()
    // axios.post.mockImplementationOnce(() => Promise.resolve(mockData));
    // const {getByText,getByTestId}=render(<BrowserRouter><SignIn></SignIn></BrowserRouter>)
    await act(async () => 
    {
    const {getByText,getByTestId}=render(<BrowserRouter><SignIn></SignIn></BrowserRouter>)
    const inputEmail = getByTestId("email",{target:{value:"abcd@gmail.com"}})
    const inputPassword = getByTestId("password",{target:{value:"abcdcom"}})
    const button=getByText("Sign In")
    fireEvent.click(button)
    await expect(axiosInstance.post).toHaveBeenCalledTimes(1)
    
});
    // const inputEmail = getByTestId("email",{target:{value:"abcd@gmail.com"}})
    // const inputPassword = getByTestId("password",{target:{value:"abcdcom"}})
    // const button=getByText("Sign In")
    // fireEvent.click(button)
    
    // form.simulate("submit")
    // expect(form.length).toBe(1)
    // await expect(axiosInstance.post).toHaveBeenCalledTimes(1)
    // await waitFor(()=>expect(axios.post).toBeCalled())
})