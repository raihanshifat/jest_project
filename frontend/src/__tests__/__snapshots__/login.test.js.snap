// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page rendered succesfully 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-1"
  >
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Sign in
    </h1>
    <h4
      style={
        Object {
          "color": "green",
        }
      }
    />
    <p
      style={
        Object {
          "color": "red",
        }
      }
    />
    <form
      className="makeStyles-form-3"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        data-testid="email"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="email"
          id="email-label"
        >
          Email Address
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="email"
            autoFocus={true}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="email"
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-7"
            >
              <span>
                Email Address
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        data-testid="password"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="password"
          id="password-label"
        >
          Password
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="current-password"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="password"
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-7"
            >
              <span>
                Password
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Sign In
        </span>
      </button>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Forgot password?
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="/signup"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Don't have an account? Sign Up
          </a>
        </div>
      </div>
    </form>
  </div>
</main>
`;
